ddl/reorg.go

```diff
@@ -727,52 +727,96 @@ func getReorgInfoFromPartitions(ctx *JobContext, d *ddlCtx, rh *reorgHandler, jo
 	return &info, nil
 }
 
-func (r *reorgInfo) UpdateReorgMeta(startKey kv.Key) error {
+func (r *reorgInfo) UpdateReorgMeta(startKey kv.Key, pool *sessionPool) (err error) {
 	if startKey == nil && r.EndKey == nil {
 		return nil
 	}
+	se, err := pool.get()
+	if err != nil {
+		return
+	}
+	defer pool.put(se)
 
-	ctx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnDDL)
-	err := kv.RunInNewTxn(ctx, r.d.store, true, func(ctx context.Context, txn kv.Transaction) error {
-		rh := newReorgHandler(meta.NewMeta(txn))
-		return errors.Trace(rh.UpdateDDLReorgHandle(r.Job, startKey, r.EndKey, r.PhysicalTableID, r.currElement))
-	})
+	sess := newSession(se)
+	err = sess.begin()
+	if err != nil {
+		return
+	}
+	txn, err := sess.txn()
 	if err != nil {
-		return errors.Trace(err)
+		sess.rollback()
+		return err
 	}
-	return nil
+	rh := newReorgHandler(meta.NewMeta(txn), sess, variable.EnableConcurrentDDL.Load())
```

Comment:
Here we use a variable(`variable.EnableConcurrentDDL.Load()`), but someplace use a value determined at startup (`w.concurrentDDL`). Does it feel like there might be some out-of-sync issues?

Reply:
if a job is running by the old framework, we should update the job, and write reorg info into the meta key, so it is determined at startup. But for backfiller, it will update the start-key using a new txn, it can be updated into the place according to the switch